### ISSUE TITLE
Beatmap for osu!taiko uses 3GB of RAM

### ISSUE THREAD

**Issue by elleyer** [ppy/osu#6123]
**Describe the crash:** 
If you trying to load this (https://osu.ppy.sh/beatmapsets/537048#osu/1138154) beatmap (taiko) on the latest difficult , osu!lazer uses 3GB of RAM, and freezes if you trying to go to the menu.
**Screenshots or videos showing encountered issue:** 
https://streamable.com/li5ze
**osu!lazer version:** 
Lastest.
**Logs:** 
[updater.log](https://github.com/ppy/osu/files/3613713/updater.log)
[network.log](https://github.com/ppy/osu/files/3613714/network.log)
[performance.log](https://github.com/ppy/osu/files/3613715/performance.log)
[runtime.log](https://github.com/ppy/osu/files/3613716/runtime.log)
**Computer Specifications:** 
G4560,
RX 470 Nitro+,
16GB DDR4-2400

**Comment by peppy** [comment]
High memory usage is something we are working on across all beatmaps. It is usually due to vertex buffer usage spiking and never being cleaned up (https://github.com/ppy/osu-framework/issues/2729).

**Comment by elleyer** [comment]
@peppy , Also I saw that after changing tracks via beatmap selector, osu! uses more RAM. Looks like tracks are loaded into RAM and never clean ( bm wasn't loaded, track only). I could record a video if needed.

**Comment by peppy** [comment]
incorrect assumption. please leave the technical side of things to us.

**Comment by smoogipoo** [comment]
Will also be fixed by #6136 if not the framework change.

**Comment by Craftplacer** [comment]
Other example beatmap: https://osu.ppy.sh/beatmapsets/758043#taiko/1594580
Tested on Windows 7 (8 GB RAM) and Windows 10 (4 GB RAM)

**Comment by bdach** [comment]
I've started work on adding pooling support for taiko, that should hopefully alleviate memory issues. I've got a branch with top-level hitobject pooling [here](https://github.com/bdach/osu/tree/taiko-pooling-main), and it seems to work OK, but I have some eldritch test failure in `TestSceneAutoplay` that I need to figure out first before PRing.

Preliminary results are promising - I tested myself on [Taiko World Cup 2016 Compilation](https://osu.ppy.sh/beatmapsets/447342#taiko/960444), and got the following results:

* current `master` - load time: 1 minute 7 seconds, memory usage delta: ~600MB,
* this PR - load time: 2 seconds, memory usage delta: ~120MB.

That is without pooling nested objects, flying hits, explosions or bar lines, so there is still room for improvements over even those numbers.
